object.c

```diff
@@ -520,7 +520,9 @@ rb_obj_size(VALUE self, VALUE args, VALUE obj)
  *
  *  Yields <i>obj</i> and returns the result.
  *
- *	'my string'.yield_self {|s|s.upcase} #=> "MY STRING"
+ *  require "ostruct"
+ *  user = OpenStruct.new(firstname: "John", lastname: "Doe")
+ *  user.yield_self { |u| "#{u.firstname} #{u.lastname}" } #=> "John Doe" 
```

Comment:
This could also be just
```ruby
"#{user.firstname} #{user.lastname}"
```
Maybe avoiding the local variable is the only way to show the purpose?
```ruby
require "ostruct"
OpenStruct.new(firstname: "John", lastname: "Doe").yield_self { |u| "#{u.firstname} #{u.lastname}" } #=> "John Doe"
# Or
OpenStruct.new(firstname: "John", lastname: "Doe").yield_self { |u|
  "#{u.firstname} #{u.lastname}"
}
# Let's abbreviate since it's so long
OpenStruct.new(first: "John", last: "Doe").yield_self { |u| "#{u.first} #{u.last}" } #=> "John Doe"
```
This example does not convince me yield_self is useful though, but that's another matter.
Maybe using a longer chain would show it more clearly.